events/publisher.go

```diff
@@ -72,12 +81,54 @@ func NewPublisher(cfg PublisherConfig) (*Publisher, error) {
 
 // PublishChange will publish a ChangeMessage to the topic for the change
 func (p *Publisher) PublishChange(ctx context.Context, subjectType string, change ChangeMessage) error {
+	ctx, span := tracer.Start(
+		ctx,
+		"events.publishChange",
+		trace.WithAttributes(
+			attribute.String(
+				"events.subject_type",
+				subjectType,
+			),
+			attribute.String(
+				"events.subject_id",
+				change.SubjectID.String(),
+			),
+			attribute.String(
+				"events.event_type",
+				change.EventType,
+			),
+			attribute.String(
+				"events.source",
+				p.source,
```

Comment:
Think we want this to be `p.Source` as I don't see the source getting set when I test this out

Reply:
`p.Source` doesn't exist; is there something else you had in mind?